Odyssey Portal

FIZLIN opened the pull request:
DEV-67

FIZLIN committed:
feat(portal): WIP

FIZLIN committed:
feat(portal): Updates to Portal logic

PAKovalev commented on packages/app/src/scenes/widgets/stores/OdysseyPortalStore/OdysseyPortalStore.ts:
```diff
@@ -0,0 +1,96 @@
+
+const ODYSSEY_WORLD_ID = 'b8b104c4-b375-4933-8de8-29e0a1c67860';
+
+const OdysseyPortalStore = types.compose(
```
It is a duplication of `OdysseyBioStore` store. We don't need to have and support two the same models.

FIZLIN replied:
✅

PAKovalev commented on packages/app/src/scenes/widgets/pages/OdysseyPortalWidget/OdysseyPortalWidget.tsx:
```diff
@@ -0,0 +1,62 @@
+
+interface PropsInterface {}
+
+const OdysseyPortalWidget: FC<PropsInterface> = () => {
```
I think we can reuse a `OdysseyInfoWidget` widget instead of creating a new one.

FIZLIN replied:
✅

PAKovalev commented on packages/app/src/scenes/widgets/Widgets.tsx:
```diff
@@ -260,6 +279,7 @@ const Widgets: FC<PropsInterface> = (props) => {
       {mutualConnectionsStore.dialog.isOpen && <MutualConnectionsWidget />}
       {onlineUsersStore.dialog.isOpen && <SearchUsersWidget />}
       {widgetsStore.odysseyBioStore.dialog.isOpen && <OdysseyBioWidget />}
+      {widgetsStore.odysseyPortalStore.dialog.isOpen && <OdysseyPortalWidget />}
```
Don't need to have this.

FIZLIN replied:
✅

PAKovalev commented on packages/app/src/scenes/widgets/Widgets.tsx:
```diff
@@ -68,6 +70,23 @@ const Widgets: FC<PropsInterface> = (props) => {
     );
   }, [agoraScreenShareStore, widgetsStore, sessionStore.userId, unityStore.worldId]);
 
+  useUnityEvent('ClickObjectEvent', (spaceId: string, label: string) => {
```
We have the hook under a `UnityPage`. Just extended that handler.

`const nft = nftStore.getNftByUuid(SOME_ID);
  if (nft) {
      widgetsStore.odysseyInfoStore.open(nft);
  }`

FIZLIN replied:
✅

FIZLIN committed:
feat(portal): Tweaks 🔧

FIZLIN committed:
feat(portal): Lint 🧹

FIZLIN committed:
feat(portal): Lint 🧹

PAKovalev commented on packages/app/src/scenes/widgets/stores/OdysseyBioStore/OdysseyBioStore.ts:
```diff
@@ -25,17 +26,22 @@ const OdysseyBioStore = types.compose(
+          self.dialog.open();
+        },
+        setOnOdysseyWorld(openingOdysseyWorld: boolean): void {
+          self.onOdysseyWorld = openingOdysseyWorld;
```
`onOdysseyWorld` what does it mean? Looks like some handler. :)

FIZLIN replied:
Fair point.

FIZLIN replied:
✅

FIZLIN committed:
feat(portal): Naming fix

FIZLIN committed:
feat(portal): Logic tweak

FIZLIN committed:
feat(portal): Disable functionalities on portal
- It was decided to remove the option to add functionalities to `Portal` objects, since they have their own portal functionality and if a user adds another one, it leads to a confusing UI

dmitry-yudakov commented on packages/app/src/scenes/widgets/stores/OdysseyBioStore/OdysseyBioStore.ts:
```diff
@@ -17,6 +17,7 @@ const OdysseyBioStore = types.compose(
       nftId: types.maybe(types.string),
       nftItem: types.maybe(types.reference(NftItem)),
       nftUser: types.maybeNull(User),
+      isOnOdysseyWorld: false,
```
Maybe it's better be a "view" function rather than model field? Checking that current nft is matching Odyssey id. ODYSSEY_WORLD_ID could be put to appVariables or maybe there's a better place for these... @PAKovalev

FIZLIN replied:
I exported the variable to the app variables and move the boolean to a view.

dmitry-yudakov reviewed (approved):
It's ok in general and could be merged, though one thing may be improved.

FIZLIN committed:
feat(portal): Moved odyssey world id to app var.
